game-core/src/main/java/games/strategy/triplea/delegate/battle/casualty/CasualtySelector.java

```diff
@@ -308,11 +298,9 @@ private static void killAmphibiousFirst(final List<Unit> killed, final Collectio
                 .combatModifiers(
                     CombatModifiers.builder()
                         .territoryEffects(territoryEffects)
-                        .amphibious(amphibious)
+                        .amphibious(targetsToPickFrom.stream().anyMatch(Unit::getWasAmphibious))
```

Comment:
I first looked at not checking for `amphibious` here and instead doing the `getWasAmphibious` check inside of the sorter.  But the sorter only looks at one unit of each type so if a battle had an amphibious marine and a non-amphibious marine, the results could be different depending on which unit was seen first.

I think the sorter needs to be looked at because, right now, if you have a partial amphibious attack (some marines are amphibious and others aren't), it sorts them as if all of the marines are amphibious.

Reply:
That would lead to an over-evaluation of those units powers, likely an outright defect.

Reply:
Do you want me to try to fix this as part of this PR?  It isn't a new issue so this change isn't making it any worse.

Reply:
new PR, bug fixes are best done as focused PRs. Ideally a test created to demonstrate the problem and then fix is relatively surgical.

One should also consider the blast radius of a roll-back. If something is wrong and there is a rollback, it's a shame to have other helpful changes also be rolled bakc.